File: TestModels/SimpleTypes/SimpleBoolean/runtimes/python-better-dirs/test/dafny/SimpleBooleanImplTest.py
import sys
from typing import Callable, Any, TypeVar, NamedTuple
from math import floor
from itertools import count

import module_
import _dafny
import System_
import Wrappers
import StandardLibrary_mUInt
import StandardLibrary
import UTF8
import simple_types_boolean_internaldafny_types
import SimpleBooleanImpl
import simple_types_boolean_internaldafny_impl

assert "SimpleBooleanImplTest" == __name__
SimpleBooleanImplTest = sys.modules[__name__]

class default__:
    def  __init__(self):
        pass

    @staticmethod
    def GetBooleanTrue():
        d_0_client_: simple_types_boolean_internaldafny_impl.SimpleBooleanClient
        d_1_valueOrError0_: Wrappers.Result = None
        out0_: Wrappers.Result
        out0_ = simple_types_boolean_internaldafny_impl.default__.SimpleBoolean(simple_types_boolean_internaldafny_impl.default__.DefaultSimpleBooleanConfig())
        d_1_valueOrError0_ = out0_
        if not(not((d_1_valueOrError0_).IsFailure())):
            raise _dafny.HaltException("test/SimpleBooleanImplTest.dfy(10,19): " + _dafny.string_of(d_1_valueOrError0_))
        d_0_client_ = (d_1_valueOrError0_).Extract()
        SimpleBooleanImplTest.default__.TestGetBooleanTrue(d_0_client_)

    @staticmethod
    def GetBooleanFalse():
        d_2_client_: simple_types_boolean_internaldafny_impl.SimpleBooleanClient
        d_3_valueOrError0_: Wrappers.Result = None
        out1_: Wrappers.Result
        out1_ = simple_types_boolean_internaldafny_impl.default__.SimpleBoolean(simple_types_boolean_internaldafny_impl.default__.DefaultSimpleBooleanConfig())
        d_3_valueOrError0_ = out1_
        if not(not((d_3_valueOrError0_).IsFailure())):
            raise _dafny.HaltException("test/SimpleBooleanImplTest.dfy(14,19): " + _dafny.string_of(d_3_valueOrError0_))
        d_2_client_ = (d_3_valueOrError0_).Extract()
        SimpleBooleanImplTest.default__.TestGetBooleanFalse(d_2_client_)

    @staticmethod
    def TestGetBooleanTrue(client):
        d_4_ret_: simple_types_boolean_internaldafny_types.GetBooleanOutput
        d_5_valueOrError0_: Wrappers.Result = Wrappers.Result_Success.default(simple_types_boolean_internaldafny_types.GetBooleanOutput.default())()
        out2_: Wrappers.Result
        out2_ = (client).GetBoolean(simple_types_boolean_internaldafny_types.GetBooleanInput_GetBooleanInput(Wrappers.Option_Some(True)))
        d_5_valueOrError0_ = out2_
        if not(not((d_5_valueOrError0_).IsFailure())):
            raise _dafny.HaltException("test/SimpleBooleanImplTest.dfy(23,16): " + _dafny.string_of(d_5_valueOrError0_))
        d_4_ret_ = (d_5_valueOrError0_).Extract()
        if not((((d_4_ret_).value).UnwrapOr(False)) == (True)):
            raise _dafny.HaltException("test/SimpleBooleanImplTest.dfy(24,8): " + _dafny.string_of(_dafny.Seq("expectation violation")))
        _dafny.print(_dafny.string_of(d_4_ret_))

    @staticmethod
    def TestGetBooleanFalse(client):
        d_6_ret_: simple_types_boolean_internaldafny_types.GetBooleanOutput
        d_7_valueOrError0_: Wrappers.Result = Wrappers.Result_Success.default(simple_types_boolean_internaldafny_types.GetBooleanOutput.default())()
        out3_: Wrappers.Result
        out3_ = (client).GetBoolean(simple_types_boolean_internaldafny_types.GetBooleanInput_GetBooleanInput(Wrappers.Option_Some(False)))
        d_7_valueOrError0_ = out3_
        if not(not((d_7_valueOrError0_).IsFailure())):
            raise _dafny.HaltException("test/SimpleBooleanImplTest.dfy(33,16): " + _dafny.string_of(d_7_valueOrError0_))
        d_6_ret_ = (d_7_valueOrError0_).Extract()
        if not((((d_6_ret_).value).UnwrapOr(True)) == (False)):
            raise _dafny.HaltException("test/SimpleBooleanImplTest.dfy(34,8): " + _dafny.string_of(_dafny.Seq("expectation violation")))
        _dafny.print(_dafny.string_of(d_6_ret_))

File: TestModels/SimpleTypes/SimpleBoolean/runtimes/go/TestsFromDafny-go/src/SimpleBooleanImplTest/SimpleBooleanImplTest.go
// Package SimpleBooleanImplTest
// Dafny module SimpleBooleanImplTest compiled into Go

package SimpleBooleanImplTest

import (
	SimpleBooleanImpl "SimpleBooleanImpl"
	StandardLibrary "StandardLibrary"
	StandardLibrary_mUInt "StandardLibrary_mUInt"
	_System "System_"
	Wrappers "Wrappers"
	_dafny "dafny"
	os "os"
	simpletypesbooleaninternaldafny "simpletypesbooleaninternaldafny"
	simpletypesbooleaninternaldafnytypes "simpletypesbooleaninternaldafnytypes"
)

var _ _dafny.Dummy__
var _ = os.Args
var _ _System.Dummy__
var _ Wrappers.Dummy__
var _ StandardLibrary_mUInt.Dummy__
var _ StandardLibrary.Dummy__
var _ SimpleBooleanImpl.Dummy__

type Dummy__ struct{}

// Definition of class Default__
type Default__ struct {
	dummy byte
}

func New_Default___() *Default__ {
	_this := Default__{}

	return &_this
}

type CompanionStruct_Default___ struct {
}

var Companion_Default___ = CompanionStruct_Default___{}

func (_this *Default__) Equals(other *Default__) bool {
	return _this == other
}

func (_this *Default__) EqualsGeneric(x interface{}) bool {
	other, ok := x.(*Default__)
	return ok && _this.Equals(other)
}

func (*Default__) String() string {
	return "SimpleBooleanImplTest.Default__"
}
func (_this *Default__) ParentTraits_() []*_dafny.TraitID {
	return [](*_dafny.TraitID){}
}

var _ _dafny.TraitOffspring = &Default__{}

func (_static *CompanionStruct_Default___) GetBooleanTrue() {
	var _74_client *simpletypesbooleaninternaldafny.SimpleBooleanClient
	_ = _74_client
	var _75_valueOrError0 Wrappers.Result = Wrappers.Result{}
	_ = _75_valueOrError0
	var _out1 Wrappers.Result
	_ = _out1
	_out1 = simpletypesbooleaninternaldafny.Companion_Default___.SimpleBoolean(simpletypesbooleaninternaldafny.Companion_Default___.DefaultSimpleBooleanConfig())
	_75_valueOrError0 = _out1
	if !(!((_75_valueOrError0).IsFailure())) {
		panic("test/SimpleBooleanImplTest.dfy(10,19): " + (_75_valueOrError0).String())
	}
	_74_client = (_75_valueOrError0).Extract().(*simpletypesbooleaninternaldafny.SimpleBooleanClient)
	Companion_Default___.TestGetBooleanTrue(_74_client)
}
func (_static *CompanionStruct_Default___) GetBooleanFalse() {
	var _76_client *simpletypesbooleaninternaldafny.SimpleBooleanClient
	_ = _76_client
	var _77_valueOrError0 Wrappers.Result = Wrappers.Result{}
	_ = _77_valueOrError0
	var _out2 Wrappers.Result
	_ = _out2
	_out2 = simpletypesbooleaninternaldafny.Companion_Default___.SimpleBoolean(simpletypesbooleaninternaldafny.Companion_Default___.DefaultSimpleBooleanConfig())
	_77_valueOrError0 = _out2
	if !(!((_77_valueOrError0).IsFailure())) {
		panic("test/SimpleBooleanImplTest.dfy(14,19): " + (_77_valueOrError0).String())
	}
	_76_client = (_77_valueOrError0).Extract().(*simpletypesbooleaninternaldafny.SimpleBooleanClient)
	Companion_Default___.TestGetBooleanFalse(_76_client)
}
func (_static *CompanionStruct_Default___) TestGetBooleanTrue(client simpletypesbooleaninternaldafnytypes.ISimpleTypesBooleanClient) {
	var _78_ret simpletypesbooleaninternaldafnytypes.GetBooleanOutput
	_ = _78_ret
	var _79_valueOrError0 Wrappers.Result = Wrappers.Companion_Result_.Default(simpletypesbooleaninternaldafnytypes.Companion_GetBooleanOutput_.Default())
	_ = _79_valueOrError0
	var _out3 Wrappers.Result
	_ = _out3
	_out3 = (client).GetBoolean(simpletypesbooleaninternaldafnytypes.Companion_GetBooleanInput_.Create_GetBooleanInput_(Wrappers.Companion_Option_.Create_Some_(true)))
	_79_valueOrError0 = _out3
	if !(!((_79_valueOrError0).IsFailure())) {
		panic("test/SimpleBooleanImplTest.dfy(23,16): " + (_79_valueOrError0).String())
	}
	_78_ret = (_79_valueOrError0).Extract().(simpletypesbooleaninternaldafnytypes.GetBooleanOutput)
	if !((((_78_ret).Dtor_value()).UnwrapOr(false).(bool)) == (true)) {
		panic("test/SimpleBooleanImplTest.dfy(24,8): " + (_dafny.SeqOfString("expectation violation")).String())
	}
	_dafny.Print(_78_ret)
}
func (_static *CompanionStruct_Default___) TestGetBooleanFalse(client simpletypesbooleaninternaldafnytypes.ISimpleTypesBooleanClient) {
	var _80_ret simpletypesbooleaninternaldafnytypes.GetBooleanOutput
	_ = _80_ret
	var _81_valueOrError0 Wrappers.Result = Wrappers.Companion_Result_.Default(simpletypesbooleaninternaldafnytypes.Companion_GetBooleanOutput_.Default())
	_ = _81_valueOrError0
	var _out4 Wrappers.Result
	_ = _out4
	_out4 = (client).GetBoolean(simpletypesbooleaninternaldafnytypes.Companion_GetBooleanInput_.Create_GetBooleanInput_(Wrappers.Companion_Option_.Create_Some_(false)))
	_81_valueOrError0 = _out4
	if !(!((_81_valueOrError0).IsFailure())) {
		panic("test/SimpleBooleanImplTest.dfy(33,16): " + (_81_valueOrError0).String())
	}
	_80_ret = (_81_valueOrError0).Extract().(simpletypesbooleaninternaldafnytypes.GetBooleanOutput)
	if !((((_80_ret).Dtor_value()).UnwrapOr(true).(bool)) == (false)) {
		panic("test/SimpleBooleanImplTest.dfy(34,8): " + (_dafny.SeqOfString("expectation violation")).String())
	}
	_dafny.Print(_80_ret)
}

// End of class Default__
